[iOS] [SaferC++] Deploy smart pointers in iOS specific code

zakariaridouh opened the pull request:
#### f9f876e65426b4ed4a8ae03aa65240ab147a3f24
<pre>
[iOS] [SaferC++] Deploy smart pointers in iOS specific code
<a href="https://bugs.webkit.org/show_bug.cgi?id=301829">https://bugs.webkit.org/show_bug.cgi?id=301829</a>
<a href="https://rdar.apple.com/163901021">rdar://163901021</a>

Reviewed by Anne van Kesteren.

Deploy smart pointers in iOS specific code.

* Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
(WebKit::AuxiliaryProcessProxy::didFinishLaunching):
* Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm:
(WebKit::ProcessAssertion::isValid const): Deleted.
* Source/WebKit/UIProcess/ProcessAssertion.cpp:
(WebKit::ProcessAssertion::isValid const): Deleted.
* Source/WebKit/UIProcess/ProcessAssertion.h:
(WebKit::ProcessAssertion::isValid const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::forwardAppBoundDomainsToITPIfInitialized):
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.cpp:
(WebKit::RemoteMediaSessionHelper::ensureProtectedConnection):
(WebKit::RemoteMediaSessionHelper::startMonitoringWirelessRoutesInternal):
(WebKit::RemoteMediaSessionHelper::stopMonitoringWirelessRoutesInternal):
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.h:

Canonical link: <a href="https://commits.webkit.org/302929@main">https://commits.webkit.org/302929@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eea01ba409b7f4c69315d96c6a9240624cc225f4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130529 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41484 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137947 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/82177 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8f21f9de-3353-4ee5-97e4-2eb7547d6525) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2812 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2692 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/99549 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67344 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2983b93b-bb13-4ead-a9de-da65af61da02) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133476 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2051 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116884 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80147 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/dcc696de-5237-4017-a8cb-1988e2ddb7c7) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1973 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81206 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110538 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35521 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140426 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2590 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/2465 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107958 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2634 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113229 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107873 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27475 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2007 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31667 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/55568 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2660 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66049 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/2479 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2681 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2586 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
```diff
@@ -379,7 +379,7 @@ void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::C
 
 #if USE(RUNNINGBOARD)
-    m_throttler.didConnectToProcess(*this);
+    Ref { m_throttler }->didConnectToProcess(*this);
```
You should use protectedThrottler() here.

zakariaridouh replied:
done.

annevk commented on Source/WebKit/UIProcess/ProcessThrottler.cpp:
```diff
@@ -295,7 +295,7 @@ void ProcessThrottler::setThrottleState(ProcessThrottleState newState)
     ProcessAssertionType newType = assertionTypeForState(newState);
 
-    if (m_assertion && m_assertion->isValid() && m_assertion->type() == newType)
+    if (m_assertion && RefPtr { m_assertion.get() }->isValid() && m_assertion->type() == newType)
```
You should inline `isValid()` instead. It's trivial. (And always true it seems...)

zakariaridouh replied:
great idea. fixed.

annevk commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
```diff
@@ -2555,7 +2555,7 @@ void WebsiteDataStore::forwardAppBoundDomainsToITPIfInitialized(CompletionHandle
     };
 
-    propagateAppBoundDomains(globalDefaultDataStore().get(), *appBoundDomains);
+    propagateAppBoundDomains(RefPtr { *globalDefaultDataStore() }.get(), *appBoundDomains);
```
Why not use protectedDefaultDataStore here?

zakariaridouh replied:
fixed!

annevk commented on Source/WebKit/UIProcess/ProcessThrottler.h:
```diff
@@ -70,7 +70,7 @@ class ProcessThrottlerActivity : public RefCountedAndCanMakeWeakPtr<ProcessThrot
     }
 
-    bool isValid() const { return !!m_throttler; }
+    inline bool isValid() const { return !!m_throttler; }
```
I don't think this is doing anything. You need to make the change in ProcessAssertion, no?

cdumez replied:
Agreed this doesn't do anything, please revert changes to this file as well.

cdumez commented on Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
```diff
@@ -346,7 +346,7 @@ void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::C
     if (launchTime > 1_s)
         RELEASE_LOG_FAULT(Process, "%s process (%p) took %f seconds to launch", processName().characters(), this, launchTime.value());
-    
+
```
Please update your IDE's settings to get rid of this noise in your PRs.

cdumez commented on Source/WebKit/UIProcess/ProcessThrottler.cpp:
```diff
@@ -45,7 +45,7 @@
 
 namespace WebKit {
-    
+
```
Please revert all changes to this file, it doesn't do anything.

cdumez commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
```diff
@@ -1192,31 +1192,31 @@ void WebsiteDataStore::isGrandfathered(const URL& url, CompletionHandler<void(bo
         return;
     }
-    
+
```
Please revert all the noise from your PR. It makes your PR more painful to review than it needs to be.

cdumez commented on Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
```diff
@@ -13,22 +13,19 @@ Platform/IPC/ArgumentCoders.h
 [ iOS ] UIProcess/API/Cocoa/_WKGeolocationPosition.mm
 [ iOS ] UIProcess/API/ios/WKWebViewIOS.mm
 [ iOS ] UIProcess/ApplicationStateTracker.mm
-[ iOS ] UIProcess/AuxiliaryProcessProxy.cpp
```
Note that we don't have EWS to validate any changes on iOS as far as I know. For this reason, I would personally hold off on unskipping anything on iOS. cc @rniwa What do you think?

zakariaridouh replied:
That's fair. We can update the expectations once the iOS EWS bot is up. Reverted changes to this file.

rniwa replied:
Sure, leaving them as is sounds good to me. I can also periodically run iOS build and report any new failures / passes.

cdumez commented on Source/WebKit/UIProcess/ProcessAssertion.h:
```diff
@@ -91,7 +91,7 @@ class ProcessAssertion : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<
     ProcessID pid() const { return m_pid; }
 
-    bool isValid() const;
+    inline bool isValid() const { return true; }
```
Please drop `inline`

annevk replied:
You don't need the inline keyword so please omit it. I think that's only necessary if the definition follows the class definition but is still in the same file.

zakariaridouh committed:
[iOS] [SaferC++] Deploy smart pointers in iOS specific code
https://bugs.webkit.org/show_bug.cgi?id=301829
rdar://163901021

Reviewed by Anne van Kesteren.

Deploy smart pointers in iOS specific code.

* Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
(WebKit::AuxiliaryProcessProxy::didFinishLaunching):
* Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm:
(WebKit::ProcessAssertion::isValid const): Deleted.
* Source/WebKit/UIProcess/ProcessAssertion.cpp:
(WebKit::ProcessAssertion::isValid const): Deleted.
* Source/WebKit/UIProcess/ProcessAssertion.h:
(WebKit::ProcessAssertion::isValid const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::forwardAppBoundDomainsToITPIfInitialized):
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.cpp:
(WebKit::RemoteMediaSessionHelper::ensureProtectedConnection):
(WebKit::RemoteMediaSessionHelper::startMonitoringWirelessRoutesInternal):
(WebKit::RemoteMediaSessionHelper::stopMonitoringWirelessRoutesInternal):
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.h:

Canonical link: https://commits.webkit.org/302929@main
